Userland/Libraries/LibJS/Runtime/GlobalObject.cpp

```diff
@@ -386,37 +386,63 @@ JS_DEFINE_NATIVE_FUNCTION(GlobalObject::parse_float)
 // 19.2.5 parseInt ( string, radix ), https://tc39.es/ecma262/#sec-parseint-string-radix
 JS_DEFINE_NATIVE_FUNCTION(GlobalObject::parse_int)
 {
+    // 1. Let inputString be ? ToString(string).
     auto input_string = TRY(vm.argument(0).to_string(global_object));
 
-    // FIXME: There's a bunch of unnecessary string copying here.
-    double sign = 1;
-    auto s = input_string.trim_whitespace(TrimMode::Left);
-    if (!s.is_empty() && s[0] == '-')
+    // 2. Let S be ! TrimString(inputString, start).
+    auto string = MUST(trim_string(global_object, js_string(vm, input_string), AK::TrimMode::Left));
+
+    // 3. Let sign be 1.
+    auto sign = 1;
+
+    // 4. If S is not empty and the first code unit of S is the code unit 0x002D (HYPHEN-MINUS), set sign to -1.
+    if (!string.is_empty() && string[0] == 0x2D)
         sign = -1;
-    if (!s.is_empty() && (s[0] == '+' || s[0] == '-'))
-        s = s.substring(1, s.length() - 1);
+    // 5. If S is not empty and the first code unit of S is the code unit 0x002B (PLUS SIGN) or the code unit 0x002D (HYPHEN-MINUS), remove the first code unit from S.
+    auto trimmed_view = string.view();
+    if (!string.is_empty() && (string[0] == 0x2B || string[0] == 0x2D))
+        trimmed_view = trimmed_view.substring_view(1);
 
+    // 6. Let R be ℝ(? ToInt32(radix)).
     auto radix = TRY(vm.argument(1).to_i32(global_object));
 
-    bool strip_prefix = true;
+    // 7. Let stripPrefix be true.
+    auto strip_prefix = true;
+
+    // 8. If R ≠ 0, then
     if (radix != 0) {
+        // a. If R < 2 or R > 36, return NaN.
         if (radix < 2 || radix > 36)
             return js_nan();
+
+        // b. If R ≠ 16, set stripPrefix to false.
         if (radix != 16)
             strip_prefix = false;
-    } else {
+    }
+    // 9. Else,
+    else {
+        // a. Set R to 10.
         radix = 10;
     }
 
+    // 10. If stripPrefix is true, then
     if (strip_prefix) {
-        if (s.length() >= 2 && s[0] == '0' && (s[1] == 'x' || s[1] == 'X')) {
-            s = s.substring(2, s.length() - 2);
+        // a. If the length of S is at least 2 and the first two code units of S are either "0x" or "0X", then
+        if (trimmed_view.length() >= 2 && trimmed_view[0] == '0' && (trimmed_view[1] == 'x' || trimmed_view[1] == 'X')) {
```

Comment:
how about this
```suggestion
        if (trimmed_view.length() >= 2 && trimmed_view.substring_view(0, 2).equals_ignoring_case("0x"sv)) {
```

or this if you want it to literally say what the spec says?

```suggestion
        if (trimmed_view.length() >= 2 && trimmed_view.substring_view(0, 2).is_any_of("0x"sv, "0X"sv)) {
```

Reply:
sure